towhee/pipeline/pipeline.py

```diff
@@ -0,0 +1,310 @@
+# pylint: disable=redefined-builtin
+
+
+class Pipeline:
+    """Pipeline is a tool to create data transformation chains.
+
+    Examples:
+    >>> from towhee.pipeline.pipeline import Pipeline
+    >>> from towhee.pipeline.action import ops
+    >>> from pprint import pprint
+    >>> def f(x):
+    ...     return x + 1
+    >>> p = ( Pipeline().input(('a', 'b'))
+    ...     .map('a', 'c', ops.towhee.decode('a', b ='b'))
+    ...     .flat_map(('a', 'b'), 'd', ops.towhee.clip('a', b ='b'))
+    ...     .output('d')
+    ... )
+    >>> print(len(p._dag))
+    4
+    """
+    #TODO: Different fn type doctest for later use
+    # >>> p = ( Pipeline().input(('a', 'b'))
+    # ...     .map('a', 'c', ops.towhee.decode('a', b ='b'))
+    # ...     .flat_map(('a', 'b'), 'd', lambda x, y: x + y)
+    # ...     .filter(('a', 'b'), ('z', 'y'), 'b', f)
+    # ...     .time_window(('a', 'b'), ('w', 'x'), 'b', 1, 1, lambda x: x > 5)
+    # ...     .batch('a', (('f', 'g'), 'h'), 1, 1, lambda x: x)
+    # ...     .batch_all('a', 'm', f)
+    # ...     .output('e')
+    # ... )
+
+    def __init__(self):
+        self._dag = {}
+
+    def set_config(self, config) -> 'Pipeline':
+        """Set the config for a pipeline.
+
+        Args:
+            config (dict): The config to set.
+
+        Returns:
+            Pipeline: Pipeline with config set.
+        """
+        self._config = config
+
+    @classmethod
+    def input(cls, output_schema) -> 'Pipeline':
+        """Start a new pipeline chain.
+
+        Args:
+            output_schema (str | tuple): The schema for the values being inputted into the pipeline.
+
+        Returns:
+            Pipeline: Pipeline ready to be chained.
+        """
+        uid = '_input'
+        pipeline = cls()
+        pipeline._dag[uid] = {
+            'input': None,
+            'output': output_schema,
+            'iter_info': {
+                'type': 'map',
+                'param': None
+            }
+        }
+        return pipeline
+
+    def output(self, input_schema, engine = 'local') -> Callable:
+        """Close and compile the pipeline chain.
+
+        Args:
+            input_schema (str | tuple): Which values to ouput.
+            engine (str, optional): Which engine to run the pipeline, currently only 'local'. Defaults to 'local'.
+
+        Returns:
+            Callable: The rendered pipeline that can be called on inputs.
+        """
+        uid = '_output'
+        self._engine = engine
+        self._dag[uid] = {
+            'input': input_schema,
+            'output': None,
+            'iter_info': {
+                'type': 'map',
+                'param': None
+            }
+        }
+        #TODO Return the DC once dc is decided.
+        return self
+
+
+    def map(self, input_schema, output_schema, fn, config = None) -> 'Pipeline':
+        """One to one map of function on inputs.
+
+        Args:
+            input_schema (str | tuple): The column/s to perform map on.
```

Comment:
Can you set the type of input_schema and output_schema to Tuple? same as other type of nodes.